taskslist/app/views/tasks/_add_task.html.erb

```diff
@@ -0,0 +1,2 @@
+<%= button_to "Add a task", "/tasks/new", :method => :get,
```

Comment:
Love all the button partials! I haven't tried using partials to this extent before, but it's definitely a solid design choice.

This is similar to the "widgets" style of web development architecture -- each self-contained piece of a page is split out into its own files and then the "page" template composes them together with appropriate variables as necessary.